libflux/src/semantic/tests.rs

```diff
@@ -255,715 +255,2377 @@ fn literals() {
-            "b" => "forall [] int",
+            "a" => "forall [] time",
         ],
-        src: r#"
-            c = a * b
-        "#,
+        src: src,
         exp: map![
-            "c" => "forall [] int",
+            "b" => "forall [] [time]",
         ],
     }
-    test_infer_err! {
+    test_infer! {
         env: map![
-            "a" => "forall [] float",
-            "b" => "forall [] int",
+            "a" => "forall [] regexp",
+        ],
+        src: src,
+        exp: map![
+            "b" => "forall [] [regexp]",
+        ],
+    }
+    test_infer! {
+        env: map![
+            "a" => "forall [] {a: int | b: float}",
+        ],
+        src: src,
+        exp: map![
+            "b" => "forall [] [{a: int | b: float}]",
+        ],
+    }
+    test_infer! {
+        env: map![
+            "a" => "forall [] {a: string | b: (x: int) -> int}",
+        ],
+        src: src,
+        exp: map![
+            "b" => "forall [] [{a: string | b: (x: int) -> int}]",
         ],
-        src: r#"
-            c = a * b
-        "#,
     }
-    //    test_infer! {
-    //        env: map![
-    //            "a" => "forall [] t0",
-    //            "b" => "forall [] int",
-    //        ],
-    //        src: r#"
-    //            c = a * b
-    //        "#,
-    //        exp: map![
-    //            "a" => "forall [] int",
-    //            "c" => "forall [] int",
-    //        ],
-    //    }
 }
 #[test]
-fn binary_expr_division() {
-    //    test_infer! {
-    //        env: map![
-    //            "a" => "forall [] t0",
-    //            "b" => "forall [] t1",
-    //        ],
-    //        src: r#"
-    //            c = a / b
+fn binary_expr_addition() {
+    // test_infer! {
+    //     env: map![
+    //         "a" => "forall [] t0",
+    //         "b" => "forall [] t1",
+    //     ],
+    //     src: r#"
+    //            c = a + b
     //        "#,
-    //        exp: map![
-    //            "a" => "forall [] t1",
-    //            "b" => "forall [] t1",
-    //            "c" => "forall [] t1",
-    //        ],
-    //    }
+    //     exp: map![
+    //         "a" => "forall [] t1",
+    //         "b" => "forall [] t1",
+    //         "c" => "forall [] t1",
+    //     ],
+    // }
     test_infer! {
         env: map![
             "a" => "forall [] int",
             "b" => "forall [] int",
         ],
         src: r#"
-            c = a * b
+            c = a + b
         "#,
         exp: map![
             "c" => "forall [] int",
         ],
     }
-    test_infer_err! {
+    test_infer! {
+        env: map![
+            "a" => "forall [] uint",
+            "b" => "forall [] uint",
+        ],
+        src: r#"
+            c = a + b
+        "#,
+        exp: map![
+            "c" => "forall [] uint",
+        ],
+    }
+    test_infer! {
         env: map![
             "a" => "forall [] float",
-            "b" => "forall [] int",
+            "b" => "forall [] float",
```

Comment:
what if you remove `b` from the prelude? Would it error or we expect infer to infer `b` too?

Reply:
It would error since `b` was not declared.